src/computer_player.c

```diff
@@ -0,0 +1,604 @@
+        if (card.name == ptr2->card.name) {
+            commonName = 1;
+        }
+        ptr2 = ptr2->next;
+    }
+    if (commonColor && commonName) {
+        return 4;
+    }
+    else if (!commonColor && commonName) {
+        return 3;
+    }
+    else if (commonColor && !commonName) {
+        return 2;
+    }
+    else if (!commonColor && !commonName) {
+        return 1;
+    }
+    return 0;
+}
+
+/*
+* @brief: function that play card from handcard list
+*
+* @return void
+*/
+void play_card(struct CARD card, struct DECK **head)
+{
+    struct DECK *prev = *head;
+    struct DECK *temp = *head;
+    // head node
+    if (temp != NULL && temp->card.name == card.name && temp->card.color == card.color)
+    {
+        *head = temp->next;
+        free(temp);
+        return;
+    }
+
+    while (temp != NULL && temp->card.name == card.name && temp->card.color == card.color)
+    {
+        prev = temp;
+        temp = temp->next;
+    }
+    // if not present
+    if (temp == NULL)
+        return;
+
+    // delete
+    prev->next = temp->next;
+
+    free(temp);
+}
+/*
+* @brief: function that check if a card exist in the handcard deck
+*
+* @return 1 exist, 0 not exist
+*/
+
+int ifexists(struct DECK **handcard, enum CardName inputCardName, enum CardColor inputCardColor)
+{
+    struct DECK *ptr = *handcard;
+    while (ptr->next != NULL)
+    {
+        if (ptr->card.name == inputCardName && ptr->card.color == inputCardColor)
+        {
+            return 1;
+        }
+        ptr = ptr->next;
+    }
+    return 0;
+}
+
+
+/*
+* @brief: function that find the largest number of given color
+*
+* @return CardName the largest number
+*/
+
+enum CardName find_largest_number(struct DECK **handcard, enum CardColor inputCardColor)
+{
+    struct DECK *temp = *handcard;
+    enum CardName Max = ZERO;
+
+
+    // Check loop while head not equal to NULL
+    while (temp != NULL)
+    {
+        
+        // If max is less then head->data then
+        // assign value of head->data to max
+        // otherwise node point to next node.
+        if (temp->card.color == inputCardColor && Max < temp->card.name)
+        {
+            Max = temp->card.name;
+        }
+        temp = temp->next;
+    }
+    
+    return Max;
+}
+
+/*
+* @brief: function that find the occruence of a number
+*
+* @return int number count for occruence
+*/
+
+int find_occurence_of_number(struct DECK **handcard, enum CardName inputCardNumber)
+{
+    struct DECK *temp = *handcard;
+    int count = 0;
+    while (temp != NULL)
+    {
+        if (temp->card.name == inputCardNumber)
+        {
+            count++;
+        }
+        temp = temp->next;
+    }
+    free(temp);
+    return count;
+}
+
+/*
+* @brief: function that find the most color occruence of a given number
+*
+* @return CardColor  most color occruence of a given number
+*
+*/
+
+enum CardColor find_color_with_most_occurence(struct DECK **handcard, enum CardName inputCardNumber)
```

Comment:
We have find_most_color() defined, the code implementation is pretty similar between, only difference is "enum CardName inputCardNumber". Why not think about to refactor two functions into a common function.